### PR TITLE
Rename token uniqueness id + Upgraded NFTs keep the same id/hash

### LINKED ISSUE #3: Rename token uniqueness id
Small issue that will reduce a lot the confusion between token id and token "uniqueness" id.

The OpenZeppelin implementation already use the term `id` for the token. Let's NOT reuse this term in the context of token.

Example of line that is super confusing:
https://github.com/QuiverCommunity/quiver-contracts/blob/6a9742d1390fe73396045f13c4e04782e21f0984/contracts/QNFT/QNFT.sol#L523

I suggest to rename the token "uniqueness" id to [hash](https://en.wikipedia.org/wiki/Hash_function):

```diff
-    mapping(uint256 => uint256) private nftIds;
+    // Mapping from token hash to token id
+    mapping(uint256 => uint256) private nftHashes;
```

Also if you want to respect the [OpenZeppelin code-style](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/bd5d681b514c9fd3e1da28e0a6c63ead9369192d/contracts/token/ERC721/ERC721Upgradeable.sol#L29-L30), I suggest:

```solidity
   // Mapping from token hash to token id
   mapping(uint256 => uint256) private _hashes;
```

This modif must be propagated to related functions' name:
```diff
-    function _getNftId(
+    function _getTokenHash(
```
```diff
-    function _setNftId(
+    function _setTokenHash(
```

Lastly, the function `nftMinted` could directly return the token id instead of a boolean. It will be easier for the DApp to fetch info related to an existing nft.This will also require to rename the function.

```diff
-    function nftMinted(
+    function tokenId(
         uint256 _imageId,
         uint256 _bgImageId,
         uint256 _favCoinId,
         uint256 _lockDuration
-    ) public view returns (bool) {
-        return
-            _getNftId(_imageId, _bgImageId, _favCoinId, _lockDuration) !=
-            uint256(0);
+    ) public view returns (uint256) {
+        return _getNftId(_imageId, _bgImageId, _favCoinId, _lockDuration);
     }
```

### LINKED ISSUE #4: Upgraded NFTs keep the same id/hash
The functions `upgradeNftImage`, `upgradeNftBackground` and `upgradeNftCoin` does not change the id/hash of the NFT that is being upgraded.

Example:
If somebody mints token with image A, then upgrades to image B, then somebody can mint the same NFT with image B.
But NOBODY can mint the token with image A.

Is it the expected flow?

I think the upgrade functions should unset the current NFT's id/hash, calculate the new one, and save it.

### LINKED ISSUE #7: defaultImageIndex is missing on mintNFT event
```
        emit MintNFT(
            msg.sender,
            nftCount,
            _imageId,
            _bgImageId,
            _favCoinId,
            _lockOptionId,
            _metaUrl
        );
 ```

### PR DESCRIPTION
**Pull request by icrabbiter**
close #3 
close #4 
close #7

### CODE REVIEW

**Comment by icrabbiter** [comment]
@NicolasMahe are you ok with this renaming?

**Review by NicolasMahe — APPROVED** [review]
looking good 👍

File: contracts/QNFT/QNFT.sol [hunk 1]
```diff
@@ -188,17 +189,21 @@ contract QNFT is
...
         return
-            _getNftId(_imageId, _bgImageId, _favCoinId, _lockDuration) !=
-            uint256(0);
+            _getTokenIdForMintOptions(
+                _imageId,
+                _bgImageId,
+                _favCoinId,
+                _lockDuration
+            );
     }
```
Comment (NicolasMahe) [inline contracts/QNFT/QNFT.sol:207]: not blocking but nice to do:
Functions `getTokenId` is just calling `_getTokenIdForMintOptions`. Please delete one

Comment (junkai121) [reply contracts/QNFT/QNFT.sol:207]: I think getTokenId is public function to be used by external

**Pull request merged**